src/App.jsx

```diff
@@ -14,33 +15,101 @@ export const goodsFromServer = [
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goodsOrdered, setGoodsOrdered] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(null);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlphabetically = () => {
+    const sorted = [...goodsOrdered].sort((a, b) => a.localeCompare(b));
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (isReversed) {
+      sorted.reverse();
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    setSortType('alphabetically');
+
+    return setGoodsOrdered(sorted);
+  };
+
+  const sortByLength = () => {
+    const sorted = [...goodsOrdered].sort(
+      (good1, good2) => good1.length - good2.length,
+    );
+
+    if (isReversed) {
+      sorted.reverse();
+    }
+
+    setSortType('length');
+
+    return setGoodsOrdered(sorted);
+  };
+
+  const reverseArray = () => {
+    setIsReversed(!isReversed);
+    const reversed = [...goodsOrdered].reverse();
+
+    return setGoodsOrdered(reversed);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const reset = () => {
+    setSortType(null);
+
+    return setGoodsOrdered(goodsFromServer);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            sortType === 'alphabetically'
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={sortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            sortType === 'length' ? 'button is-info' : 'button is-info is-light'
+          }
+          onClick={sortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            isReversed ? 'button is-success' : 'button is-success is-light'
+          }
+          onClick={reverseArray}
+        >
+          Reverse
+        </button>
+
+        {goodsOrdered !== goodsFromServer ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={reset}
+          >
+            Reset
+          </button>
+        ) : null}
+      </div>
+
+      <ul>
+        {goodsOrdered.map(good => (
+          <li data-cy="Good">{good}</li>
```

Comment:
Camille, this `<li>` tag is missing a `key` property.
Whenever we use some type of iteration to generate multiple elements, like this `.map` here, we need to pass an unique key for each element, so React knows which is which.
In this case, you can use `key={good}` like so:
```
<li data-cy="Good" key={good}>
    {good}
</li>
```

Reply:
Ah sim, eu tinha posto utilizando o index, mas o eslint não tava deixando eu subir, ai eu tirei já que não tem ID. Nem tinha pensando em usar o proprio good como Key